otb-luci/files/usr/lib/lua/luci/controller/overthebox.lua

```diff
@@ -33,3 +35,28 @@ function otb_dhcp_leases_status()
   luci.http.prepare_content("application/json")
   luci.http.write_json(s.dhcp_leases())
 end
+
+function _readr(src)
+   local dst = {}
+   local type = nixio.fs.stat(src, "type")
+   if type == "dir" then
+      for file in nixio.fs.dir(src) do
+         dst[file] = _readr(src..'/'..file);
```

Comment:
It might be better to write two separate functions: one to read directories and one to read a file.
There's no need for a recursive function here.